doc: clarify `useCodeCache` setting for cross-platform SEA generation (#53994)
nodejs/node: doc/api/single-executable-applications.md

current working directory. The version of the Node.js binary used to produce
the blob must be the same as the one to which the blob will be injected.

Note: When generating cross-platform SEAs (e.g., generating an SEA on
platform A for platform B), `useCodeCache` and `useSnapshot` must be set to

> false to avoid generating 'broken' executables. The generated executable might
> crash on startup if these settings are not configured properly.

Reviewer: ```suggestion
false to avoid generating incompatible executables. Since code cache and snapshots can only be loaded
on the same platform where they are compiled, the generated executable might crash on startup when
trying to load code cache or snapshots built on a different platform.
```

current working directory. The version of the Node.js binary used to produce
the blob must be the same as the one to which the blob will be injected.

Note: When generating cross-platform SEAs (e.g., generating an SEA on

> platform A for platform B), `useCodeCache` and `useSnapshot` must be set to

Reviewer: ```suggestion
for `linux-x64` on `darwin-arm64`), `useCodeCache` and `useSnapshot` must be set to
```

> Note: When generating cross-platform SEAs (e.g., generating an SEA on
> for `linux-x64` on `darwin-arm64`), `useCodeCache` and `useSnapshot` must be set to
> false to avoid generating incompatible executables. Since code cache and snapshots can only be loaded
> on the same platform where they are compiled, the generated executable might crash on startup when
> trying to load code cache or snapshots built on a different platform.

Reviewer 2: ```suggestion
Note: When generating cross-platform SEAs (e.g., generating a SEA
for `linux-x64` on `darwin-arm64`), `useCodeCache` and `useSnapshot` must be set to
false to avoid generating incompatible executables. Since code cache and snapshots can only be loaded
on the same platform where they are compiled, the generated executable might crash on startup when
trying to load code cache or snapshots built on a different platform.
```

current working directory. The version of the Node.js binary used to produce
the blob must be the same as the one to which the blob will be injected.

> Note: When generating cross-platform SEAs (e.g., generating an SEA on platform A for platform B), `useCodeCache` must be set to false to avoid generating ‘broken’ executables. The generated executable might crash on startup if this setting is not configured properly.

Reviewer 3: ```suggestion
Note: When generating cross-platform SEAs (e.g., generating an SEA on
platform A for platform B), `useCodeCache` must be set to false to avoid
generating ‘broken’ executables. The generated executable might crash
on startup if this setting is not configured properly.
```
